pkg/controllers/workgenerator/controller.go

```diff
@@ -180,23 +180,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 			Message:            "All of the works are synchronized to the latest",
 		})
 		if workUpdated {
-			// revert the applied condition and failedPlacement if we made any changes to the work
+			// revert the applied condition, available condition, and failedPlacement if we made any changes to the work
 			resourceBinding.Status.FailedPlacements = nil
-			resourceBinding.SetConditions(metav1.Condition{
-				Status:             metav1.ConditionFalse,
-				Type:               string(fleetv1beta1.ResourceBindingApplied),
-				Reason:             condition.WorkNeedSyncedReason,
-				Message:            "In the processing of synchronizing the work to the member cluster",
-				ObservedGeneration: resourceBinding.Generation,
-			})
+			//resourceBinding.SetConditions(metav1.Condition{
+			//	Status:             metav1.ConditionFalse,
+			//	Type:               string(fleetv1beta1.ResourceBindingApplied),
+			//	Reason:             condition.WorkNeedSyncedReason,
+			//	Message:            "In the processing of synchronizing the work to the member cluster",
+			//	ObservedGeneration: resourceBinding.Generation,
+			//})
+			//resourceBinding.SetConditions(metav1.Condition{
+			//	Status:             metav1.ConditionFalse,
+			//	Type:               string(fleetv1beta1.ResourceBindingAvailable),
+			//	Reason:             condition.WorkNeedSyncedReason,
+			//	Message:            "In the processing of synchronizing the work to the member cluster",
+			//	ObservedGeneration: resourceBinding.Generation,
+			//})
+			resourceBinding.RemoveCondition(string(fleetv1beta1.ResourceBindingApplied))
```

Comment:
Just a nit: we could set the status to unknown if necessary. It would satisfy both the semantics and our use case.

Reply:
since the resource binding has changed and we haven't applied it yet, I think it's safe to say that the new resource applied/available is "false".  "Unknown" works too. Removing probably also works semantically.